MSBLOC.Core/Services/CheckRunSubmitter.cs

```diff
@@ -37,22 +38,41 @@ public class CheckRunSubmitter : ICheckRunSubmitter
                 };
 
                 return newCheckRunAnnotation;
-            }).Take(50).ToList();
+            }).Batch(50).ToArray();
+
+            var checkConclusion = buildDetails.Annotations
+                .Any(annotation => annotation.AnnotationWarningLevel == AnnotationWarningLevel.Failure) ? CheckConclusion.Failure : CheckConclusion.Success;
 
             var newCheckRun = new NewCheckRun(checkRunName, headSha)
             {
                 Output = new NewCheckRunOutput(checkRunTitle, checkRunSummary)
                 {
-                    Annotations = newCheckRunAnnotations
+                    Annotations = newCheckRunAnnotations.FirstOrDefault()?.ToArray()
                 },
                 Status = CheckStatus.Completed,
                 StartedAt = startedAt,
                 CompletedAt = completedAt,
-                Conclusion = buildDetails.Annotations
-                    .Any(annotation => annotation.AnnotationWarningLevel == AnnotationWarningLevel.Failure) ? CheckConclusion.Failure : CheckConclusion.Success
+                Conclusion = checkConclusion
             };
 
-            return await CheckRunsClient.Create(owner, name, newCheckRun);
+            var checkRun = await CheckRunsClient.Create(owner, name, newCheckRun);
+
+            foreach (var newCheckRunAnnotation in newCheckRunAnnotations.Skip(1))
```

Comment:
Any reason we're processing these synchronously?

why not:
`await Task.WhenAll(newCheckRunAnnotations.Skip(1).Select(newCheckRunAnnotation => CheckRunsClient.Update(...)));`

Reply:
Wanna just kick this can up the road and put an issue in the backlog?